pkg/election/election.go

```diff
@@ -349,10 +391,50 @@ func (e *Election) watchLeader(ctx context.Context, session *concurrency.Session
 			return
 		case <-ctx.Done():
 			return
+		case <-e.resignCh:
+			if e.evictLeader.Get() == 1 {
```

Comment:
Can we remove this line and others can call close(e.resignCh) to resign leader as well?